Fw/Types/FileNameString.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Fw/Types/FileNameString.hpp"
+#include "Fw/Types/StringUtils.hpp"
+
+namespace Fw {
+
+FileNameString::FileNameString(const char* src) : StringBase() {
+    (void)Fw::StringUtils::string_copy(this->m_buf, src, sizeof(this->m_buf));
+}
+
+FileNameString::FileNameString(const StringBase& src) : StringBase() {
+    (void)Fw::StringUtils::string_copy(this->m_buf, src.toChar(), sizeof(this->m_buf));
+}
+
+FileNameString::FileNameString(const FileNameString& src) : StringBase() {
+    (void)Fw::StringUtils::string_copy(this->m_buf, src.toChar(), sizeof(this->m_buf));
+}
+
+FileNameString::FileNameString() : StringBase() {
+    this->m_buf[0] = 0;
+}
+
+FileNameString& FileNameString::operator=(const FileNameString& other) {
+    if (this == &other) {
+        return *this;
+    }
+
+    (void)Fw::StringUtils::string_copy(this->m_buf, other.toChar(), sizeof(this->m_buf));
+    return *this;
+}
+
+FileNameString& FileNameString::operator=(const StringBase& other) {
+    if (this == &other) {
+        return *this;
+    }
+
+    (void)Fw::StringUtils::string_copy(this->m_buf, other.toChar(), sizeof(this->m_buf));
```

Comment:
We want to use the size of the destination buffer, to avoid overrunning the buffer. For example, a string of size 100 that's copied into a buffer of size 40 will be truncated.

Reply:
You're right on that part that we want to limit the max to be the local buffer size. How about a minimum of the source actual string size and the local storage size?

Reply:
The string copy utility does do a minimum. It copies the actual length of the source string unless that length is greater than the buffer size, in which case it truncates the string to fit in the buffer size. This is a standard size-bounded copy operation. It's just like strncpy, except that it's safer.